crates/pop-cli/src/commands/call/contract.rs

```diff
@@ -435,12 +474,55 @@ impl CallContractCommand {
 		}
 	}
 
+	/// Execute the smart contract call using wallet integration.
+	async fn execute_with_secure_signing(
+		&self,
+		call_exec: CallExec<DefaultConfig, DefaultEnvironment, Keypair>,
+		cli: &mut impl Cli,
+	) -> Result<()> {
+		let call_data = self.get_contract_data(&call_exec).map_err(|err| {
+			anyhow!("An error occurred getting the call data: {}", err.to_string())
+		})?;
+
+		let maybe_payload = wait_for_signature(call_data, self.url.to_string()).await?;
+		if let Some(payload) = maybe_payload {
+			cli.success("Signed payload received.")?;
+			let spinner = spinner();
+			spinner.start("Calling the contract...");
```

Comment:
This is never cancelled, leaving an unnecessary artifact on screen.

Reply:
https://github.com/r0gue-io/pop-cli/pull/371/commits/ddb8ba6db338bfe3e1c6d091c726e65afdea9f09#diff-60233806b7c6493fe1742695952147d4dc1bcefcbdb4886da81e63e788941bddR420